### PR TITLE
Add Compression with Associated Documentation

### PR DESCRIPTION
**Pull request by tianhuil**
Implementing https://github.com/scrapy/scrapy/issues/2174

### CODE REVIEW

File: docs/topics/feed-exports.rst [hunk 1]
```diff
@@ -258,6 +259,15 @@ which uses safe numeric encoding (``\uXXXX`` sequences) for historic reasons.
...
 Use ``utf-8`` if you want UTF-8 for JSON too.
 
+.. setting:: FEED_EXPORT_COMPRESSION
+
+FEED_EXPORT_COMPRESSION
+-----------------------
+
+Default: ``False``
+
+Set to ``True`` to enable GZip compression.
```
Comment (Gallaecio) [inline docs/topics/feed-exports.rst]: :lipstick: Maybe we should call it ‘gzip’ and link to https://en.wikipedia.org/wiki/Gzip for reference.

File: scrapy/extensions/feedexport.py [hunk 2]
```diff
@@ -204,6 +204,7 @@ def __init__(self, settings):
             raise NotConfigured
         self.format = settings['FEED_FORMAT'].lower()
         self.export_encoding = settings['FEED_EXPORT_ENCODING']
+        self.export_compression = settings.get('FEED_EXPORT_COMPRESSION', False)
```
Comment (Gallaecio) [inline scrapy/extensions/feedexport.py]: I _think_ adding the default value to `default_settings.py` is enough, no need to add it here as well. And `Settings.getbool` should be used to ensure that changing the setting from environment variables, the command line and such works the same.

```suggestion
        self.export_compression = settings.getbool('FEED_EXPORT_COMPRESSION')
```

**Comment by elacuesta** [comment]
Should we merge #3858 first and rebase this on top of it?

**Comment by wRAR** [comment]
@tianhuil the multi-feed PR is now merged, feel free to continue working on this.

**Comment by Koen87** [comment]
This method doesn't work combined with the S3 Feed Storage. I've been trying to make it work myself with the code changes in this PR but haven't had any success yet.

**Comment by Gallaecio** [comment]
Implemented in https://github.com/scrapy/scrapy/pull/5190

**Pull request closed without merging**